src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -0,0 +1,1249 @@
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.embeddings.tok_embeddings
+
+    def set_input_embeddings(self, value):
+        self.embeddings.tok_embeddings = value
+
+    def _init_weights(self, module: Optional[nn.Module] = None, reset_params: Optional[bool] = None):
+        if module and hasattr(module, "_init_weights"):
+            super()._init_weights(module, reset_params)
+        elif isinstance(reset_params, bool):
+            self.embeddings._init_weights(reset_params=reset_params)
+
+            if reset_params:
+                self.final_norm.reset_parameters()
+
+    def forward(
+        self,
+        input_ids: torch.LongTensor = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        return_dict: Optional[bool] = None,
+        indices: Optional[torch.Tensor] = None,
+        cu_seqlens: Optional[torch.Tensor] = None,
+        max_seqlen: Optional[int] = None,
+        batch_size: Optional[int] = None,
+        seq_len: Optional[int] = None,
+    ) -> Union[Tuple[torch.Tensor, ...], BaseModelOutput]:
+        if batch_size is None and seq_len is None:
+            batch_size, seq_len = input_ids.shape[:2]
+
+        if attention_mask is None:
+            attention_mask = torch.ones((batch_size, seq_len), device=input_ids.device, dtype=torch.bool)
+
+        repad = False
+        if self.config.unpad_inputs:
+            if indices is None and cu_seqlens is None and max_seqlen is None:
+                repad = True
+                if self.config.unpad_no_grad:
+                    input_ids, indices, cu_seqlens, max_seqlen, *_ = self._unpad_inputs_no_grad(
+                        input_ids, attention_mask
+                    )
+                else:
+                    input_ids, indices, cu_seqlens, max_seqlen, *_ = self._unpad_inputs(input_ids, attention_mask)
+        elif position_ids is None:
+            position_ids = torch.arange(seq_len, device=input_ids.device)
+
+        hidden_states = self.embeddings(input_ids)
+
+        for encoder_layer in self.layers:
+            if self.gradient_checkpointing and self.training:
+                hidden_states = self._gradient_checkpointing_func(
+                    encoder_layer.__call__,
+                    hidden_states,
+                    attention_mask,
+                    position_ids,
+                    cu_seqlens,
+                    max_seqlen,
+                )
+            else:
+                hidden_states = encoder_layer(
+                    hidden_states,
+                    attention_mask=attention_mask,
+                    position_ids=position_ids,
+                    cu_seqlens=cu_seqlens,
+                    max_seqlen=max_seqlen,
+                )
+
+        hidden_states = self.final_norm(hidden_states)
+
+        if repad:
+            hidden_states = self._pad_outputs(hidden_states, indices, batch_size, seq_len)
```

Comment:
```suggestion
            hidden_states, _ = self._pad_outputs(hidden_states, indices, batch_size, seq_len)
```
This also returns padded `labels`, which we don't use.

Reply:
fixed